class-reservation-1.php

```diff
@@ -0,0 +1,121 @@
+                id mediumint(9) NOT NULL AUTO_INCREMENT,
+                time datetime DEFAULT '0000-00-00 00:00:00' NOT NULL,
+                name tinytext NOT NULL,
+                last_name tinytext NOT NULL,
+                event_id bigint(20) NOT NULL,
+                user_id bigint(20) NOT NULL,
+                PRIMARY KEY  (id)
+              ) $charset_collate;";
+              
+              require_once( ABSPATH . 'wp-admin/includes/upgrade.php' );
+              dbDelta( $sql );
+        }
+
+        function reservation_form () {
+            ?>
+            <h2>RSVP</h2>
+            <form method="post">
+            <div>
+            <label for="name"><?php esc_html_e('First Name','eventspractice'); ?></label>
+            <input type="text" name="name" id="name" value="" required="" aria-required="true">
+            </div>
+            <div>
+            <label for="name"><?php esc_html_e('Last Name','eventspractice'); ?></label>
+            <input type="text" name="last_name" id="last_name" value="" required="" aria-required="true">
+            </div>
+            <div>
+            <label for="email"><?php esc_html_e('Email','eventspractice'); ?></label>
+            <input type="email" name="email" id="email" value="" required="" aria-required="true">
+            </div>
+            <?php wp_nonce_field( basename( __FILE__ ), 'eventspractice_reservation_nonce' ); ?>
+            <button class="wp-block-button__link" data-id-attr="placeholder" type="submit"><?php esc_html_e('Send reservation','eventspractice'); ?></button>
+            </form>
+            <?php
+        }
+        
+        
+        function complete_reservation( $name, $last_name, $event_id, $user_id ) {
+                // $args = [
+                //     'post_title'    => get_the_title() . " — Reserved by " . $username,
+                //     'post_status'   => 'publish',
+                //     'post_type'     => 'reservation',
+                //     'post_author'   => $author,
+                // ];
+                // wp_insert_post($args);
+                
+                global $wpdb;
+
+                $table = $wpdb->prefix.'eventspractice_reservations';
+                
+                $wpdb->insert(
+                    $table,
+                    array (
+                        'name'      => $name,
+                        'last_name' => $last_name,
+                        'event_id'  => $event_id,
+                        'user_id'   => $user_id,
+                    ),
+                    array (
+                        '%s', 
+                        '%s', 
+                        '%d', 
+                        '%d'
+                    )
+                );
+
+                echo '<p><strong>' . esc_html__('Reservation completed.','eventspractice') . '</strong></p>';
+                return true;   
+        }
+        
+        function custom_reservation_function() {
```

Comment:
Makes sense, thank you!

I'll definitely refactor this since this is a modified code from an example I found. I'll save this for the next iteration since I'll be using plugin template parts instead of the shortcode logic, and will take that opportunity to check this code and rethink the approach :)